crates/pathfinder/src/state/sync/l1.rs

```diff
@@ -1,680 +1,21 @@
-
-/// Events and queries emitted by L1 sync process.
-#[derive(Debug)]
-pub enum Event {
-    /// New L1 [update logs](StateUpdateLog) found.
-    Update(Vec<StateUpdateLog>),
-    /// An L1 reorg was detected, contains the reorg-tail which
-    /// indicates the oldest block which is now invalid
-    /// i.e. reorg-tail + 1 should be the new head.
-    Reorg(StarknetBlockNumber),
-    /// Query for the [update log](StateUpdateLog) of the given block.
-    ///
-    /// The receiver should return the [update log](StateUpdateLog) using the
-    /// [oneshot::channel].
-    QueryUpdate(StarknetBlockNumber, oneshot::Sender<Option<StateUpdateLog>>),
-}
-
-/// Syncs L1 state update logs. Emits [sync events](Event) which should be handled
-/// to update storage and respond to queries.
-pub async fn sync<T>(
-    tx_event: mpsc::Sender<Event>,
-    transport: T,
-    chain: Chain,
-    core_address: H160,
-    head: Option<StateUpdateLog>,
-) -> anyhow::Result<()>
-where
-    T: EthereumTransport + Send + Sync + Clone,
-{
-    let eth_api = EthereumImpl {
-        logs: Arc::new(RwLock::new(StateRootFetcher::new(
-            head,
-            chain,
-            core_address.0.into(),
-        ))),
-        transport,
-    };
-
-    // The core sync logic implementation.
-    sync_impl(eth_api, tx_event, chain).await
-}
-
-#[cfg_attr(test, mockall::automock)]
-#[async_trait::async_trait]
-trait EthereumApi {
-    async fn fetch_logs(&mut self) -> Result<Vec<StateUpdateLog>, FetchError>;
-
-    async fn set_log_head(&mut self, head: Option<StateUpdateLog>);
-
-    async fn log_head(&self) -> Option<StateUpdateLog>;
-
-    async fn block_hash(
-        &self,
-        block: EthereumBlockNumber,
-    ) -> anyhow::Result<Option<EthereumBlockHash>>;
-}
-
-/// A helper function to keep the backoff strategy construction separated.
-async fn retry<T, E, Fut, FutureFactory, RetryCondition>(
-    future_factory: FutureFactory,
-    retry_condition: RetryCondition,
-) -> Result<T, E>
-where
-    Fut: Future<Output = Result<T, E>>,
-    FutureFactory: FnMut() -> Fut,
-    RetryCondition: FnMut(&E) -> bool,
-{
-    Retry::exponential(future_factory, NonZeroU64::new(2).unwrap())
-        .factor(NonZeroU64::new(15).unwrap())
-        .max_delay(Duration::from_secs(10 * 60))
-        .when(retry_condition)
-        .await
-}
-
-#[derive(Clone)]
-struct EthereumImpl<T: EthereumTransport + Send + Sync> {
-    logs: Arc<RwLock<StateRootFetcher>>,
-    transport: T,
-}
-
-#[async_trait::async_trait]
-impl<T: EthereumTransport + Send + Sync + Clone> EthereumApi for EthereumImpl<T> {
-    async fn fetch_logs(&mut self) -> Result<Vec<StateUpdateLog>, FetchError> {
-        let ff = || async {
-            let logs = self.logs.clone();
-            let transport = self.transport.clone();
-            let mut logs = logs.write().await;
-            logs.fetch(transport).await
-        };
-        let logs = retry(ff, |error| match error {
```

Comment:
iiuc there's no retry/backoff strategy in case of L1 right? just restart the l1 process after and minute?

Reply:
I'm not also sure we need one honestly, 60s backoff every time seems good enough, @Mirko-von-Leipzig you did all of the earlier interacting with L1, wdyt?

Reply:
Oh, one more thing - an L1 error which previously was ignored/hidden in the logs because L1 did a retry will now get exposed as this scary "L1 process restarted" log which could freak ppl out :thinking:

Reply:
We should probably auto-retry + backoff on 429 and some 500s yeah.